src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -255,11 +255,22 @@ private bool VisitChildren(IValidationStrategy strategy)
             return isValid;
         }
 
-        private IList<IModelValidator> GetValidators(ModelMetadata metadata)
+        private IList<IModelValidator> GetValidators()
         {
-            var context = new ModelValidatorProviderContext(metadata);
-            _validatorProvider.GetValidators(context);
-            return context.Validators.OrderBy(v => v, ValidatorOrderComparer.Instance).ToList();
+            if (_context == null)
+            {
+                _context = new ModelValidatorProviderContext(_metadata);
+            }
+            else
+            {
+                // Reusing the context to improve performance.
```

Comment:
You can do better than this comment. You need to mention that without reusing the context we'd allocate a list + context for every model property that we validate.